python/demo/gmsh/demo_gmsh.py

```diff
@@ -5,33 +5,101 @@
 # SPDX-License-Identifier:    LGPL-3.0-or-later
```

Comment:
Are the functions in this file general enough to be included as part of the library, and not only in this demo?

Reply:
I think they eventually should be part of the library. I put them here, so that we can discuss how we would like to interact with the gmsh API.

Reply:
Great. Do you and @garth-wells think that these two functions can have `gmsh.model` or `gmsh.model.mesh` as input? This would allow avoiding `import gmsh` in `dolfinx.io`.
(BTW, how would one create two meshes at the same time with such gmsh api?)

Reply:
We could put the model name as input, (then using Gmsh.model.setCurrent) such that it sets the current model, exports the geometry and topology, and returns the mesh.

Reply:
> Great. Do you and @garth-wells think that these two functions can have `gmsh.model` or `gmsh.model.mesh` as input? This would allow avoiding `import gmsh` in `dolfinx.io`.
> 
> (BTW, how would one create two meshes at the same time with such gmsh api?)

However, i am not sure how one would take a Gmsh model as input.

Reply:
Let's see how it shakes down and then decide if it can go somewhere in the library. Maybe rather than going deep into the core code we could have a `dolfinx.utilities` module or something similar.

Reply:
> Let's see how it shakes down and then decide if it can go somewhere in the library. Maybe rather than going deep into the core code we could have a `dolfinx.utilities` module or something similar.

If we do not make a type hint for the gmsh model, the only import of gmsh now is when reading in msh files.

Reply:
Both `dolfinx.io` and `dolfinx.utilities` sound good to me.